### ISSUE TITLE
Inability to Fetch Multiple Nodes

### ISSUE THREAD

**Issue by cocoahero** [Shopify/mobile-buy-sdk-ios#691]
The Storefront GraphQL schema appears to support fetching multiple nodes at once with the following query:

```
{
  nodes(ids: ["idOne", "idTwo"]) {
    ... on Collection {
      id
      title
    }
  }
}
```

However, no equivalent API appears to be available in this SDK.

**Comment by dbart01** [comment]
That's correct, the SDK does not currently support the `nodes` field. However, a workaround exists using [this approach](https://github.com/Shopify/mobile-buy-sdk-ios#aliases-).

**Comment by cocoahero** [comment]
Unfortunately, this workaround doesn't appear to work for fetching a dynamic amount of nodes of a given type. 

For example, given array of collection IDs, to use aliases would mean having to generate index based aliases, given the position of the ID in the array, like such:

```swift
let query = Storefront.buildQuery({
    for (index, id) in collectionIDs.enumerated() {
        $0.node(alias: "collection-\(index)", id: id, { $0
            .onCollection(subfields: { $0
                .id()
                .title()
                .image({ $0
                    .src()
                })
            })
        })
    }
})
```

In theory, retrieving the collections back out would simply be a matter of re-iterating and re-using the generated aliases, like so:

```swift
var collections = Array<Storefront.Collection>()
for (index, _) in collectionIDs.enumerated() {
    if let collection = response.aliasedNode(alias: "collection-\(index)") as? Storefront.Collection {
        collections.append(collection)
    }
}
return collections
```

However, this doesn't appear to work because the GraphQL query parser fails when the alias contains unexpected characters.

```swift
invalidQuery([Buy.Graph.QueryError.Reason(message: "Parse error on \"-0\" (INT) at [1, 18]", line: nil, column: nil)])```

**Comment by cocoahero** [comment]
Although, after writing that update, removing the `-` between `collection` and the index appears to work fine.  I am still concerned, however, with the fragility of such an implementation.

**Comment by dbart01** [comment]
I agree, the workaround is less than ideal. We are, however, looking into implementing native `nodes` query support in the SDK.

**Comment by dbart01** [comment]
Now available in the [3.1 release](https://github.com/Shopify/mobile-buy-sdk-ios/tree/3.1).
